ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/requesthandler/internalresponse/EthAccountsBodyProvider.java

```diff
@@ -30,10 +29,10 @@
 
   private static final Logger LOG = LogManager.getLogger();
 
-  private final String address;
+  private final Collection<String> addresses = new HashSet<>();
 
-  public EthAccountsBodyProvider(final String address) {
-    this.address = address;
+  public EthAccountsBodyProvider(final Collection<String> address) {
+    this.addresses.addAll(address);
```

Comment:
Won't this just return the initial set of addresses for a signer? So if the addresses change in the future we would be returning the old list of addresses for the eth_getAccounts?

Reply:
Ouch! Nice catch! We should probably make this also ask the provider for the available addresses.